build error on Windows due to CreatePalette name collision

**Describe**
When trying to build DevilutionX on Windows, using MSys2 and Mingw64, I get the following error:

```
C:\tools\msys64\mingw64\x86_64-w64-mingw32\include\wingdi.h:2709:29: error: functions that differ only in their return type cannot be overloaded
  WINGDIAPI HPALETTE WINAPI CreatePalette(CONST LOGPALETTE *plpal);
            ~~~~~~~~        ^
./Source/dx.h:16:6: note: previous declaration is here
void CreatePalette();
~~~~ ^
```

**To Reproduce**
Steps to reproduce the behavior:
1. Clone latest source of DevilutionX
2. Build using MSys2 and Mingw64 on Windows 10.
3. See build error.

**Expected behavior**
Successful compilation.

**Additional context**
I think we had this issue in the past, related to `random`, `SetCursor` and some other functions that collide with the Windows API.

--- a/Source/dx.h
+++ b/Source/dx.h
@@ -13,7 +13,7 @@ void unlock_buf(BYTE idx);
 void dx_cleanup();
 void dx_reinit();
 
-void CreatePalette();
+void InitPalette();
 void BltFast(SDL_Rect *src_rect, SDL_Rect *dst_rect);
 void Blit(SDL_Surface *src, SDL_Rect *src_rect, SDL_Rect *dst_rect);
 void RenderPresent();

--- a/Source/palette.cpp
+++ b/Source/palette.cpp
@@ -70,7 +70,7 @@ void palette_init()
 {
 	LoadGamma();
 	memcpy(system_palette, orig_palette, sizeof(orig_palette));
-	CreatePalette();
+	InitPalette();
 }
 
 void LoadPalette(char *pszFileName)

--- a/SourceX/dx.cpp
+++ b/SourceX/dx.cpp
@@ -164,7 +164,7 @@ void dx_reinit()
 	force_redraw = 255;
 }
 
-void CreatePalette()
+void InitPalette()
 {
 	palette = SDL_AllocPalette(256);
 	if (palette == NULL) {

